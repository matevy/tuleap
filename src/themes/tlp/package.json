{
  "author": "Enalean Team",
  "name": "@tuleap/tlp",
  "description": "Tuleap CSS framework",
  "license": "GPL-2.0-or-later",
  "main": "index.js",
  "types": "./src/index.d.ts",
  "private": true,
  "dependencies": {
    "@formatjs/intl-getcanonicallocales": "^1.5.1",
    "@formatjs/intl-numberformat": "^5.7.1",
    "@formatjs/intl-pluralrules": "^3.5.1",
    "@formatjs/intl-relativetimeformat": "^7.3.1",
    "@webcomponents/custom-elements": "^1.4.2",
    "flatpickr": "^4.6.6",
    "jquery": "^3.5.1",
    "popper.js": "^1.16.1",
    "select2": "^4.0.13"
  },
  "devDependencies": {
    "codemirror": "^5.57.0",
    "dompurify": "^2.2.2"
  },
  "config": {
    "bin": "../../../node_modules/.bin"
  },
  "scripts": {
    "test": "$npm_package_config_bin/jest"
  }
}
