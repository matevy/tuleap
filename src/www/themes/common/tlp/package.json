{
  "name": "tlp",
  "version": "0.0.1",
  "description": "Tuleap CSS framework",
  "author": "Enalean",
  "license": "GPL-2.0-or-later",
  "main": "index.js",
  "private": true,
  "dependencies": {
    "core-js": "^3.2.0",
    "dom4": "^2.1.5",
    "flatpickr": "^4.6.2",
    "jquery": "2.1.0",
    "popper.js": "^1.14.7",
    "regenerator-runtime": "^0.13.3",
    "select2": "^4.0.10",
    "whatwg-fetch": "^3.0.0"
  },
  "devDependencies": {
    "@types/select2": "^4.0.48",
    "codemirror": "^5.48.4",
    "dompurify": "^2.0.7"
  },
  "types": "./src/index.d.ts",
  "config": {
    "bin": "../../../../../node_modules/.bin"
  },
  "scripts": {
    "build": "$npm_package_config_bin/webpack --config webpack.prod.js",
    "test": "$npm_package_config_bin/jest",
    "watch": "$npm_package_config_bin/concurrently '$npm_package_config_bin/webpack --config webpack.dev.js --watch' 'CI=true npm test -- --watch'"
  }
}
