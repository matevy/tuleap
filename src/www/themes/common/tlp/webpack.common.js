/*
 * Copyright (c) Enalean, 2017-Present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

const path = require("path");
const polyfills_for_fetch = require("../../../../../tools/utils/scripts/ie11-polyfill-names.js")
    .polyfills_for_fetch;
const webpack_configurator = require("../../../../../tools/utils/scripts/webpack-configurator.js");

let entry_points = {
    "tlp-en_US": polyfills_for_fetch.concat(["dom4", "./src/index.en_US.js"]),
    "tlp-fr_FR": polyfills_for_fetch.concat(["dom4", "./src/index.fr_FR.js"])
};

const colors = ["orange", "blue", "green", "red", "grey", "purple"];
for (const color of colors) {
    entry_points[`tlp-${color}`] = `./src/scss/tlp-${color}.scss`;
    entry_points[`tlp-${color}-condensed`] = `./src/scss/tlp-${color}-condensed.scss`;
}

const tlp_framework_config = {
    entry: entry_points,
    context: path.resolve(__dirname),
    output: {
        path: path.resolve(__dirname, "dist/"),
        filename: "tlp-[chunkhash].[name].js",
        library: "tlp"
    },
    resolve: {
        alias: {
            select2: "select2/dist/js/select2.full.js"
        }
    },
    module: {
        rules: [
            webpack_configurator.configureBabelRule(webpack_configurator.babel_options_ie11),
            webpack_configurator.rule_scss_loader,
            webpack_configurator.rule_css_assets
        ]
    },
    plugins: [
        webpack_configurator.getCleanWebpackPlugin(),
        webpack_configurator.getManifestPlugin(),
        ...webpack_configurator.getCSSExtractionPlugins()
    ]
};

const tlp_doc_config = {
    entry: {
        style: "./doc/css/main.scss",
        script: "./doc/js/index.js"
    },
    context: path.resolve(__dirname),
    output: webpack_configurator.configureOutput(path.resolve(__dirname, "doc/dist/")),
    externals: {
        tlp: "tlp"
    },
    module: {
        rules: [
            webpack_configurator.configureBabelRule(webpack_configurator.babel_options_ie11),
            webpack_configurator.rule_scss_loader
        ]
    },
    plugins: [
        webpack_configurator.getCleanWebpackPlugin(),
        webpack_configurator.getManifestPlugin(),
        ...webpack_configurator.getCSSExtractionPlugins()
    ]
};

module.exports = [tlp_framework_config, tlp_doc_config];
