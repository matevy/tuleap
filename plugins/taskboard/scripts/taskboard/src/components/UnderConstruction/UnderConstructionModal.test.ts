/*
 * Copyright (c) Enalean, 2019 - present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import * as tlp from "tlp";
import { shallowMount, Wrapper } from "@vue/test-utils";
import {
    createStoreMock,
    Store
} from "../../../../../../../src/www/scripts/vue-components/store-wrapper-jest";
import { createTaskboardLocalVue } from "../../helpers/local-vue-for-test";
import UnderConstructionModal from "./UnderConstructionModal.vue";

jest.mock("tlp", () => {
    return {
        __esModule: true,
        modal: jest.fn()
    };
});

async function createWrapper<G, Spy>(
    store: Store<TestState, G, Spy>
): Promise<Wrapper<UnderConstructionModal>> {
    return shallowMount(UnderConstructionModal, {
        localVue: await createTaskboardLocalVue(),
        mocks: {
            $store: store
        }
    });
}

interface TestState {
    user: {
        user_id: number;
    };
}

const storage_key = "tuleap-taskboard-under-construction-modal-hidden-104";

describe("UnderConstructionModal", () => {
    let store: Store<TestState>;
    const tlp_modal = (tlp.modal as unknown) as jest.SpyInstance;
    let hide_listener: Function;
    beforeEach(() => {
        store = createStoreMock({ state: { user: { user_id: 104 } } });
        tlp_modal.mockReset();
        tlp_modal.mockImplementation(() => {
            return {
                addEventListener: jest
                    .fn()
                    .mockImplementation((event_name: string, listener: Function) => {
                        hide_listener = listener;
                    }),
                show: jest.fn()
            };
        });
        sessionStorage.clear();
    });

    it(`When the modal has not been opened before,
        it will open the modal as soon as the component is mounted`, async () => {
        const actual_tlp = jest.requireActual("tlp");
        jest.spyOn(tlp, "modal").mockImplementation(actual_tlp.modal);

        const wrapper = await createWrapper(store);

        expect(wrapper.element).toMatchSnapshot();
    });

    it(`When the modal is closed, it will save the current date/time`, async () => {
        await createWrapper(store);
        hide_listener();

        const saved_date = sessionStorage.getItem(storage_key);
        expect(saved_date).not.toBeNull();
    });

    it(`When the modal has been opened in the last 24 hours,
        it won't open the modal`, async () => {
        const six_hours_ago = new Date();
        six_hours_ago.setHours(six_hours_ago.getHours() - 6);
        sessionStorage.setItem(storage_key, six_hours_ago.toUTCString());

        await createWrapper(store);

        expect(tlp_modal).not.toHaveBeenCalled();
    });

    it(`When the modal has been opened before the last 24 hours,
        it will open the modal again`, async () => {
        const thirty_six_hours_ago = new Date();
        thirty_six_hours_ago.setDate(thirty_six_hours_ago.getDate() - 1);
        thirty_six_hours_ago.setHours(thirty_six_hours_ago.getHours() - 12);
        sessionStorage.setItem(storage_key, thirty_six_hours_ago.toUTCString());

        await createWrapper(store);

        expect(tlp_modal).toHaveBeenCalled();
    });

    it(`When I'm browsing as anonymous user,
        the session storage key will be suffixed by "0"`, async () => {
        store.state.user.user_id = 0;
        await createWrapper(store);
        hide_listener();

        const saved_date = sessionStorage.getItem(
            "tuleap-taskboard-under-construction-modal-hidden-0"
        );
        expect(saved_date).not.toBeNull();
    });
});
