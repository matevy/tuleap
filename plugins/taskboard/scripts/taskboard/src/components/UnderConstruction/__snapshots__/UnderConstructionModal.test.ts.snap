// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnderConstructionModal When the modal has not been opened before,
        it will open the modal as soon as the component is mounted 1`] = `
<div
  aria-labelledby="taskboard-under-construction-modal-title"
  class="tlp-modal tlp-modal-warning tlp-modal-display tlp-modal-shown"
  role="dialog"
>
  <div
    class="tlp-modal-header"
  >
    <h1
      class="tlp-modal-title"
      id="taskboard-under-construction-modal-title"
    >
      <i
        class="fa fa-warning tlp-modal-title-icon"
      />
       
      <translate-stub
        tag="span"
      >
        This feature is under construction
      </translate-stub>
    </h1>
     
    <div
      class="tlp-modal-close"
      data-dismiss="modal"
      role="button"
      title="Close"
    >
      
            ×
        
    </div>
  </div>
   
  <div
    class="tlp-modal-body"
  >
    <translate-stub
      tag="p"
    >
      Hello,
    </translate-stub>
     
    <p>
      Thanks for your interest in Tuleap next features. You're about to test a new feature that is under construction. All the buttons, links, menus, … might not work. This application 
      <strong>
        will not work on IE11
      </strong>
      .
    </p>
     
    <p>
      If you have any questions or if you want to leave feedback, drop an email to 
      <a
        href="mailto:feedback-taskboard@enalean.com"
      >
        feedback-taskboard@enalean.com
      </a>
       or join us on 
      <a
        href="https://chat.tuleap.org"
      >
        chat.tuleap.org
      </a>
      .
    </p>
  </div>
   
  <div
    class="tlp-modal-footer"
  >
    <button
      class="tlp-button-warning tlp-modal-action"
      data-dismiss="modal"
      type="button"
    >
      <i
        class="fa fa-thumbs-up tlp-button-icon"
      />
       
      <translate-stub
        tag="span"
      >
        I understand this is under construction
      </translate-stub>
    </button>
  </div>
</div>
`;
